app/presenters/hyrax/homepage_presenter.rb

```diff
@@ -6,8 +6,19 @@ def initialize(current_ability)
       @current_ability = current_ability
     end
 
+    # @return [Boolean] If the current user is a guest and the always_display_share_button
+    #   is activated, then return true. Otherwise return true if the signed in
+    #   user has permission to create at least one kind of work.
     def display_share_button?
-      Hyrax.config.always_display_share_button? || current_ability.can_create_any_work?
+      (user_unregistered? && Hyrax.config.always_display_share_button?) ||
```

Comment:
I'm a bit confused by the logic, is it the same as the following? If so, I prefer the following as it flows better when reading. There is a nuanced difference in that the `Hyrax.config.always_display_share_button?` has a higher order of precedence.

The logic as it stands seems to say only care about the `always_display_share_button?` if we have an unregistered user.

```ruby
return true if Hyrax.config.always_display_share_button?
return false if user_unregistered?
current_ability.can_create_any_work?
```